modin/pandas/dataframe.py

```diff
@@ -167,9 +167,11 @@ def __init__(
             if columns is not None and not isinstance(columns, pandas.Index):
                 columns = pandas.Index(columns)
             if columns is not None:
-                self = self.set_axis(columns, axis=1, copy=False)
+                obj_with_new_columns = self.set_axis(columns, axis=1, copy=False)
+                self._query_compiler = obj_with_new_columns._query_compiler
```

Comment:
We have a method `._update_inplace()` which is aimed at updating in-place, I would think it makes sense to use that (as it's doing a little more than just updating a QC)